test/hotspot/gtest/gtestMain.cpp

```diff
@@ -112,27 +110,37 @@ class JVMInitializerListener : public ::testing::EmptyTestEventListener {
   int _argc;
   char** _argv;
   bool _is_initialized;
+  JavaVM* jvm;
 
   void initialize_jvm() {
   }
 
  public:
   JVMInitializerListener(int argc, char** argv) :
-    _argc(argc), _argv(argv), _is_initialized(false) {
+    _argc(argc), _argv(argv), _is_initialized(false), jvm(NULL) {
   }
 
   virtual void OnTestStart(const ::testing::TestInfo& test_info) {
     const char* name = test_info.name();
     if (!_is_initialized && is_same_vm_test(name)) {
       // we want to have hs_err and core files when we execute regular tests
-      int ret_val = init_jvm(_argc, _argv, false);
+      int ret_val = init_jvm(_argc, _argv, false, &jvm);
```

Comment:
init_jvm should just return the JavaVM pointer instead of a (rather boolean) int. The result int does not carry any information beyond "success". We can do the same with != NULL.

Reply:
I think the right thing to do is to report the JNI error that gets returned.